pennylane_lightning/lightning_qubit_new.py

```diff
@@ -0,0 +1,136 @@
+from pennylane import QubitStateVector, BasisState, DeviceError
+
+from ._version import __version__
+
+
+class LightningQubitNew(DefaultQubit):
+    """PennyLane Lightning device.
+
+    An extension of PennyLane's built-in ``default.qubit`` device that interfaces with C++ to
+    perform fast linear algebra calculations
+
+    Use of this device requires pre-built binaries or compilation from source. Check out the
+    :doc:`/installation` guide for more details.
+
+    Args:
+        wires (int): the number of wires to initialize the device with
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to 1000 if not specified.
+            If ``analytic == True``, then the number of shots is ignored
+            in the calculation of expectation values and variances, and only controls the number
+            of samples returned by ``sample``.
+        analytic (bool): indicates if the device should calculate expectations
+            and variances analytically
+    """
+
+    name = "Lightning Qubit PennyLane plugin"
+    short_name = "lightning.qubit"
+    pennylane_requires = ">=0.12"
+    version = __version__
+    author = "Xanadu Inc."
+
+    operations = {
+        "BasisState",
+        "QubitStateVector",
+        "PauliX",
+        "PauliY",
+        "PauliZ",
+        "Hadamard",
+        "S",
+        "T",
+        "CNOT",
+        "SWAP",
+        "CSWAP",
+        "Toffoli",
+        "CZ",
+        "PhaseShift",
+        "RX",
+        "RY",
+        "RZ",
+        "Rot",
+        "CRX",
+        "CRY",
+        "CRZ",
+        "CRot",
+    }
+
+    observables = {"PauliX", "PauliY", "PauliZ", "Hadamard", "Identity"}
+
+    def __init__(self, wires, *, shots=1000, analytic=True):
+        super().__init__(wires, shots=shots, analytic=analytic)
+
+    @classmethod
+    def capabilities(cls):
+        capabilities = super().capabilities().copy()
+        capabilities.update(
+            model="qubit",
+            supports_reversible_diff=False,
+            supports_inverse_operations=False,
+            supports_analytic_computation=True,
+            returns_state=True,
+        )
+        return capabilities
+
+    def apply(self, operations, rotations=None, **kwargs):
+        for i, operation in enumerate(operations):  # State preparation is currently done in Python
+            if isinstance(operation, (QubitStateVector, BasisState)):
+                if i == 0:
+                    self._apply_operation(operation)
+                    del operations[0]
+                else:
+                    raise DeviceError(
+                        "Operation {} cannot be used after other Operations have already been "
+                        "applied on a {} device.".format(operation.name, self.short_name)
+                    )
+
+        if operations:
+            # XXX: Do we need a copy here? Not sure of the required copy semantics
```

Comment:
you do not need a copy here.

Reply:
Actually I take that back I'm not sure. @antalszava do we need to use `np.copy()` here?